### PR TITLE
Added support for IPv6

### PR DESCRIPTION
**Pull request by cmyers-spirent**
This change is [Reviewable](https://reviewable.io/reviews/spirentorion/osv/3)

### CODE REVIEW

File: bsd/sys/compat/linux/linux_socket.h [hunk 1]
```diff
@@ -151,4 +151,64 @@ struct l_ucred {
 #define	LINUX_IP_ADD_MEMBERSHIP		35
 #define	LINUX_IP_DROP_MEMBERSHIP	36
 
+
+/* Options for use with `getsockopt' and `setsockopt' at the IPv6 level.
+   The first word in the comment at the right is the data type used;
```
Comment (DerangedMonkeyNinja) [inline bsd/sys/compat/linux/linux_socket.h]: What does the comment after {get, set}stockpot apply to?  Am I blind?

File: include/api/netinet6/__in6.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2013 Cloudius Systems, Ltd.
```
Comment (DerangedMonkeyNinja) [inline include/api/netinet6/__in6.h]: Umm...  wrong copyright, I think...

File: bsd/sys/netinet6/raw_ip6.cc [hunk 3]
```diff
@@ -664,6 +663,15 @@ rip6_attach(struct socket *so, int proto, struct thread *td)
 		free(filter);
 		return (ENOMEM);
 	}
+	/* soreserve() must be done faer inpcb i screated which sets up the socket so_mtx */
```
Comment (DerangedMonkeyNinja) [inline bsd/sys/netinet6/raw_ip6.cc]: Umm....

File: bsd/sys/compat/linux/linux_socket.h (hunk 1 above)
Comment (cmyers-spirent) [reply bsd/sys/compat/linux/linux_socket.h]: You are right.  Looks like that comment is obsolete.  Not sure where it came from...

File: include/api/netinet6/__in6.h (hunk 2 above)
Comment (cmyers-spirent) [reply include/api/netinet6/__in6.h]: File was originally copied from include/api/netinet/__in.h and includes that copyright.
Should I take it out completely or just update to 2018?

**Pull request merged**